include/hipSYCL/runtime/signal_channel.hpp

```diff
@@ -51,14 +53,13 @@ class signal_channel {
   }
 
   bool has_signalled() const {
-    auto future = _shared_future;
-    return future.wait_for(std::chrono::seconds(0)) ==
-           std::future_status::ready;
+    return(has_signalled_flag.load());
```

Comment:
`atomic<T>` has conversion operators and assignment operators for `T`, so you don't need to use explicit `load()` and `store()`.